core/domain/user_jobs_one_off.py

```diff
@@ -468,6 +469,170 @@ def reduce(key, values):
         yield (key, len(values))
 
 
+class FixUserSettingsCreatedOnOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that fixes the invalid values of created_on attribute in the
+    UserSettingsModel. It is a one-off job and can be removed from the codebase
+    after we resolve this issue by running the job once.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(FixUserSettingsCreatedOnOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_settings_model):
+        user_id = user_settings_model.id
+        user_dates_list = [
+            (
+                'UserSettingsModel_last_updated',
+                user_settings_model.last_updated
+            ),
+            (
+                'UserSettingsModel_last_agreed_to_terms',
+                user_settings_model.last_agreed_to_terms
+            ),
+            (
+                'UserSettingsModel_last_started_state_editor_tutorial',
+                user_settings_model.last_started_state_editor_tutorial
+            ),
+            (
+                'UserSettingsModel_last_started_state_translation_tutorial',
+                user_settings_model.last_started_state_translation_tutorial
+            ),
+            (
+                'UserSettingsModel_last_logged_in',
+                user_settings_model.last_logged_in
+            ),
+            (
+                'UserSettingsModel_last_edited_an_exploration',
+                user_settings_model.last_edited_an_exploration
+            ),
+            (
+                'UserSettingsModel_last_created_an_exploration',
+                user_settings_model.last_created_an_exploration
+            ),
+        ]
+
+        if user_settings_model.first_contribution_msec is not None:
+            user_dates_list.append(
+                (
+                    'UserSettingsModel_first_contribution_msec',
+                    datetime.datetime.fromtimestamp(
+                        python_utils.divide(
+                            user_settings_model.first_contribution_msec, 1000.0)
+                    )
```

Comment:
Don't we have some code for this transformation somewhere already, maybe in utils?

Reply:
No, I had already checked for it in `utils.py`, but this function is not present. We have similar functions (like converting string to datetime object etc) but not timestamp to . 
Is it fine here or do you want me to create one function in `utils.py` and then use it here? (We only want to run the job once, but such function could still be useful).

Reply:
@vojtechjelinek Yes I did check for it `utils.py`, but a function for this specific transformation does not exist as of now there. We have related functions, like converting string to datetime object etc, but not this one.

Is it okay the current way or does it make more sense to create a function in `utils.py` and then use it here (if you think it could be useful elsewhere, because this job only needs to be run once).

Reply:
Let's keep it here then.

Reply:
Sure!